src/wp-includes/class-wp-scripts.php

```diff
@@ -836,6 +821,11 @@ public function add_data( $handle, $key, $value ) {
 			return false;
 		}
 
+		if ( 'conditional' === $key ) {
+			// If a dependency is declared by a conditional script, remove it.
+			$this->registered[ $handle ]->deps = array();
+		}
```

Comment:
this should only be the case if the conditional script is the _only_ dependency pulling it in, right?

or at this point in the code will that be handled because the deps here are only for the condition script itself and not _all scripts_?

Reply:
There are details here: https://github.com/WordPress/wordpress-develop/pull/9497/files/f2011cdc95a9369a5e6e865e2e004b8593a97ff3#diff-65a2f96e438d914ebca50c9eb6b80dca7ab942fbd2ae80491af9da9db0bc32ff